### PR TITLE
#7 Update customDistributionVersion property to optional with project.version value as default

### PR DESCRIPTION
**Pull request by serhiiluhovyi**

### CODE REVIEW

File: src/main/kotlin/tech/harmonysoft/oss/gradle/dist/CustomGradleDistributionPlugin.kt [hunk 1]
```diff
@@ -9,16 +9,16 @@ class CustomGradleDistributionPlugin : Plugin<Project> {
...
-            validateAndEnrich(config)
+            validateAndEnrich(config, project)
             project.tasks.register("buildGradleDist", BuildCustomGradleDistributionTask::class.java) {
                 it.config.set(config)
             }
         }
     }
 
-    private fun validateAndEnrich(config: CustomGradleDistConfig) {
+    private fun validateAndEnrich(config: CustomGradleDistConfig, project: Project) {
```
Comment (denis-zhdanov) [inline src/main/kotlin/tech/harmonysoft/oss/gradle/dist/CustomGradleDistributionPlugin.kt:21]: we don't need to provide `Project` as a parameter here - it's already available as a property, other methods already use that approach

File: src/main/kotlin/tech/harmonysoft/oss/gradle/dist/CustomGradleDistributionPlugin.kt [hunk 2]
```diff
@@ -39,18 +39,22 @@ class CustomGradleDistributionPlugin : Plugin<Project> {
         }
     }
 
-    private fun validateCustomDistributionVersion(config: CustomGradleDistConfig) {
-        if (!config.customDistributionVersion.isPresent || config.customDistributionVersion.get().isBlank()) {
+    private fun configureCustomDistributionVersionIfNecessary(config: CustomGradleDistConfig, project: Project) {
+        val customDistributionVersionNotSpecified = !config.customDistributionVersion.isPresent || config.customDistributionVersion.get().isBlank()
```
Comment (denis-zhdanov) [inline src/main/kotlin/tech/harmonysoft/oss/gradle/dist/CustomGradleDistributionPlugin.kt]: this is a local variable, we can use less verbose name like `notSpecified`

File: src/test/kotlin/tech/harmonysoft/oss/gradle/dist/CustomGradleDistributionPluginTest.kt [hunk 3]
```diff
@@ -198,7 +198,76 @@ class CustomGradleDistributionPluginTest {
...
+                "build/gradle-dist/gradle-$GRADLE_VERSION-$PROJECT_NAME-$customDistributionVersion-all.zip"
+        )
+        verifyDistributionFile(expectedCustomDistributionFile)
+    }
+
+    @Test
+    fun `when customDistributionVersion and project version are not defined then build fails`() {
+        try {
+            doTest("single-distribution-no-templates", """
+            plugins {
```
Comment (denis-zhdanov) [inline src/test/kotlin/tech/harmonysoft/oss/gradle/dist/CustomGradleDistributionPluginTest.kt]: missing indentation

**Comment by denis-zhdanov** [comment]
finished review iteration

**Comment by serhiiluhovyi** [comment]
ready for review

**Comment by piotrminkina** [comment]
@serhii-luhovyi  The implementation is much more complicated than it could be. All you have to do is create a convention for this extension, and Gradle will do all the work for you. Add the following code directly after the extension implementation and voila.

```kotlin
config.customDistributionVersion.convention(
    project.provider { project.version.toString() }
)
```

**Comment by denis-zhdanov** [comment]
superseded by https://github.com/harmonysoft-tech/custom-gradle-dist-gradle-plugin/pull/15

**Pull request closed without merging**